warehouse/query-core/src/main/java/datawave/query/attributes/ExcerptFields.java

```diff
@@ -112,19 +113,47 @@ public boolean containsField(String field) {
      * @return the offset
      */
     public Integer getOffset(String field) {
-        return fieldMap.get(field);
+        return fieldMap.get(field).firstKey();
     }
 
     /**
-     * Put the offset to use when retrieving excerpts for the specified field.
+     * Return the direction to use when retrieving excerpts for the specified field.
+     *
+     * @param field
+     *            the field
+     * @return the direction
+     */
+    public String getDirection(String field) {
+        SortedMap<Integer,String> offsetMap = fieldMap.get(field);
+        return offsetMap.get(getOffset(field));
+    }
+
+    /**
+     * Put the offset to use when retrieving excerpts for the specified field. When no direction is specified, we default to returning both directions.
      *
      * @param field
      *            the field
      * @param offset
      *            the offset
      */
     public void put(String field, Integer offset) {
-        fieldMap.put(field, offset);
+        this.put(field, offset, "both");
```

Comment:
Other spot for a potential constant here.

Reply:
I agree, a constant for "both" would be goodness

Reply:
Aye aye